editor/actions.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+import { parse, stringify } from 'querystring';
+
+export function savePost( dispatch, postId, edits ) {
+	const toSend = postId ? { id: postId, ...edits } : edits;
+	const isNew = ! postId;
+
+	dispatch( {
+		type: 'REQUEST_POST_UPDATE',
+		edits,
+		isNew,
+	} );
+
+	new wp.api.models.Post( toSend ).save().done( ( newPost ) => {
+		dispatch( {
+			type: 'REQUEST_POST_UPDATE_SUCCESS',
+			post: newPost,
+			isNew,
+		} );
+		if ( isNew && window.history.replaceState ) {
+			const urlPieces = window.location.href.split( '?' );
+			const qs = parse( urlPieces[ 1 ] || '' );
+			const newUrl = urlPieces[ 0 ] + '?' + stringify( {
+				...qs,
```

Comment:
Minor: The object spread syntax creates another clone step; this is the equivalent of:

```js
Object.assign( {}, qs, { post_id: newPost.id } );
```

Since we've created and have no plans to reuse the object outside the scope of the function, I think it'd be fine enough to mutate `qs` by assigning directly into it:

```js
Object.assign( qs, { post_id: newPost.id } );
```